### PR TITLE
Web rendering pipeline separation

### PR DESCRIPTION
**Pull request by WojciechBarczynski**

### CODE REVIEW

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            render_pass.set_bind_group(2, &self.sampler.bind_group, &[]);
+
+            self.wgpu_ctx
+                .plane_cache
+                .plane(input_id)
+                .unwrap()
+                .draw(&mut render_pass);
+        };
+
+        if common_params.texture_count == 0 {
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:188]: this does not really make sense for web renderer, you will never have zero textures

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                    push_constant_ranges: &[wgpu::PushConstantRange {
+                        stages: wgpu::ShaderStages::VERTEX_FRAGMENT,
+                        range: 0..CommonShaderParameters::push_constant_size(),
+                    }],
+                });
+
+        let pipeline = wgpu_ctx
+            .device
+            .create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+                label: Some("shader transformation pipeline :^)"),
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:67]: change label

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        })
+    }
+
+    pub(super) fn render(
+        &self,
+        params: &wgpu::BindGroup,
+        textures: &[Option<&Texture>],
+        target: &NodeTextureState,
+        pts: Duration,
+        clear_color: Option<wgpu::Color>,
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:117]: you don't use this option

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            pipeline,
+            textures_bgl,
+            sampler,
+        })
+    }
+
+    pub(super) fn render(
+        &self,
+        params: &wgpu::BindGroup,
+        textures: &[Option<&Texture>],
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:114]: I would change the signature of this method to fit the web renderer, e.g. separately pass texture from web renderer and textures from children, also modify shader code.

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        target: &NodeTextureState,
+        pts: Duration,
+        clear_color: Option<wgpu::Color>,
+    ) {
+        let mut texture_views: Vec<&wgpu::TextureView> = textures
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:119]: make the same change as in the layout node, and pass only one texture per draw call. If you want to do that in follow-up PR it's fine, but I think there is not much point in separating that.

File: compositor_render/src/transformations/web_renderer/shader.rs [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            sampler,
+        })
+    }
+
+    pub(super) fn render(
+        &self,
+        params: &wgpu::BindGroup,
+        textures: &[Option<&Texture>],
+        target: &NodeTextureState,
+        pts: Duration,
```
Comment (wkozyra95) [inline compositor_render/src/transformations/web_renderer/shader.rs:116]: you don't need pts here

**Comment by WojciechBarczynski** [comment]
Almost all suggestions are already applied in the follow-up PR. I will merge all those PRs to a separate branch and create the one PR with all changes bundled together.

**Pull request merged**